### PR TITLE
added support for data units [issue #4]

### PR DESCRIPTION
**Pull request by julianofischer**
Added support for data units as discussed in issue #4.

### CODE REVIEW

**Comment by julianofischer** [comment]
Because it is trivial and a typo, I fixed an error at line 114 of NodeDensityReport.

**Comment by tk721** [comment]
It's not actually a typo, it's valid Java 7. I guess we don't have a policy on the supported Java version going forward, but it's good to keep Java 6 support for now if it's just these kinds of trivial incompatibilities.

**Comment by julianofischer** [comment]
You are totally right, Teemu.
My mistake. Sorry.

I  also think it is a good idea keep Java 6 support.
Indeed, when using Java 7 we are prone to the [this](http://www.coderanch.com/t/587592/java/java/sort-TimSort-MergeSort-java-Dangerous) error, related to Collections.sort method. The easiest workaround is to use the old merge sort from Java 6 by setting a JVM property: 

    -Djava.util.Arrays.useLegacyMergeSort=true.

File: src/report/NodeDensityReport.java [hunk 1]
```diff
@@ -111,7 +111,7 @@ public NodeDensityReport() {
 		final double duration = scenario.getEndTime();
 		final int sampleCount = (int) (duration / super.interval + 1);
 
-		this.samples = new ArrayList<>(sampleCount);
+		this.samples = new ArrayList<int[][]>(sampleCount);
```
Comment (akeranen) [inline src/report/NodeDensityReport.java]: Should this change be reverted then?

**Comment by tk721** [comment]
It should be left as `new ArrayList<int[][]>` so that it compiles with Java 6. Unless we deprecate Java 6, which doesn't really make sense unless we want/need some features from 7 or 8.

**Comment by akeranen** [comment]
OK, so then we're good to go. I'll merge this. Thanks guys!

**Pull request merged**